WebClient/doc/core/modelCollection/modelCollection.md

```diff
@@ -0,0 +1,38 @@
+# ModelCollection
+
+*ModelCollection* is a storage of models. This storage emits events whenever items are added or removed.
+
+## API reference
+
+Extends the [L.Evented](https://leafletjs.com/reference-1.0.0.html#evented) class therefore it provides convenient subscription methods like *on(...)* and *off(...)*.
+
+#### Options
+Options object that should be passed to *ModelCollection* constructor.
+
+| Property | Type | Description |
+|---|---|---|
+| models | Array&lt;object&gt; | *Optional*. Initial array containing model objects. **Note** that models should have an `id` property. |
+
+#### Properties
+
+| Property | Type | Description |
+|---|---|---|
+| models | Array&lt;object&gt; | Array containing model objects. Changing this array will not have an effect on the actual property but changing the model objects will actually change them.  |
+
+#### Methods
+
+| Method | Returns | Description |
+|---|---|---|
+| set(Array&lt;object&gt; models) | void | Sets the models to the collection. Previously stored models will be removed from this collection. `change` event is emitted. If models are removed from the collection then `remove` event is emitted. |
```

Comment:
Is the change event emitted per model removed?

Reply:
Make more specific..

Reply:
One event per function call. This means if multiple models are removed within a single `set()` call then one `change` event and one `remove` event will be emitted. `Change` event will provide the new state of the collection while `remove` event will provide the list of removed models.

Reply:
I'll extend the description to be more precise